systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -1460,6 +1463,69 @@ void checkStorageSizeForVolumes(List<PersistentVolumeClaim> volumes, String[] di
         }
     }
 
+    @Tag(SCALABILITY)
+    @Test
+    void testBigAmountOfTopics() {
+        int numberOfTopics = 300;
+
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1).done();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String topicName = "topic-example" + i;
+            testMethodResources().topic(CLUSTER_NAME, topicName, 3, 3).done();
+            LOGGER.info("Checking that topic {} was created", topicName);
+            assertThat(cmdKubeClient().list("kafkatopic"), hasItems(topicName));
+        }
+    }
+
+    @Test
+    void testRegenerateCertExternalAddressChange() throws InterruptedException {
+        LOGGER.info("Creating kafka without external listener");
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1).done();
+
+        List<Secret> secretsWithoutExt = kubeClient().listSecrets();
+
+        LOGGER.info("Editing kafka with external listener");
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 3, 1)
+            .editSpec()
+                .editKafka()
+                    .editListeners()
+                        .withNewKafkaListenerExternalLoadBalancer()
+                        .endKafkaListenerExternalLoadBalancer()
+                    .endListeners()
+                .endKafka()
+            .endSpec()
+            .done();
+
+        LOGGER.info("Waiting for secrets change");
+        StUtils.waitUntilSecretsChange();
+
+        List<Secret> secretsWithExt = kubeClient().listSecrets();
+        List<Secret> brokerSecrets = new ArrayList<>(2);
+
+        for (Secret secret : secretsWithoutExt) {
+            if (secret.getMetadata().getName().equals("my-cluster-kafka-brokers")) {
```

Comment:
If we are interested only in one particular secret, why are you using the list? Why not use the `kubeClient` to get the particular secret directly?

Reply:
Yes, of course.